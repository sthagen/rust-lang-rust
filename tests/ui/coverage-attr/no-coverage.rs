//@ reference: attributes.coverage.allowed-positions

#![feature(extern_types)]
#![feature(coverage_attribute)]
#![feature(impl_trait_in_assoc_type)]
#![warn(unused_attributes)]
#![coverage(off)]

#[coverage(off)] //~ ERROR attribute should be applied to a function definition or closure
trait Trait {
    #[coverage(off)] //~ ERROR attribute should be applied to a function definition or closure
    const X: u32;

    #[coverage(off)] //~ ERROR attribute should be applied to a function definition or closure
    type T;

    type U;

    #[coverage(off)] //~ ERROR attribute should be applied to a function definition or closure
    fn f(&self);

    #[coverage(off)] //~ ERROR attribute should be applied to a function definition or closure
    fn g();
}

#[coverage(off)]
impl Trait for () {
    const X: u32 = 0;

    #[coverage(off)] //~ ERROR attribute should be applied to a function definition or closure
    type T = Self;

    #[coverage(off)] //~ ERROR attribute should be applied to a function definition or closure
    type U = impl Trait; //~ ERROR unconstrained opaque type

    fn f(&self) {}
    fn g() {}
}

extern "C" {
    #[coverage(off)] //~ ERROR attribute should be applied to a function definition or closure
    static X: u32;

    #[coverage(off)] //~ ERROR attribute should be applied to a function definition or closure
    type T;
}

#[coverage(off)]
fn main() {
    #[coverage(off)] //~ ERROR attribute should be applied to a function definition or closure
    let _ = ();

    match () {
        #[coverage(off)] //~ ERROR attribute should be applied to a function definition or closure
        () => (),
    }

    #[coverage(off)] //~ ERROR attribute should be applied to a function definition or closure
    return ();
}
