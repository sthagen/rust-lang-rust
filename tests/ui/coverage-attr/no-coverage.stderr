error[E0788]: attribute should be applied to a function definition or closure
  --> $DIR/no-coverage.rs:9:1
   |
LL |   #[coverage(off)]
   |   ^^^^^^^^^^^^^^^^
LL | / trait Trait {
LL | |     #[coverage(off)]
LL | |     const X: u32;
...  |
LL | |     fn g();
LL | | }
   | |_- not a function or closure

error[E0788]: attribute should be applied to a function definition or closure
  --> $DIR/no-coverage.rs:50:5
   |
LL |     #[coverage(off)]
   |     ^^^^^^^^^^^^^^^^
LL |     let _ = ();
   |     ----------- not a function or closure

error[E0788]: attribute should be applied to a function definition or closure
  --> $DIR/no-coverage.rs:54:9
   |
LL |         #[coverage(off)]
   |         ^^^^^^^^^^^^^^^^
LL |         () => (),
   |         -------- not a function or closure

error[E0788]: attribute should be applied to a function definition or closure
  --> $DIR/no-coverage.rs:58:5
   |
LL |     #[coverage(off)]
   |     ^^^^^^^^^^^^^^^^
LL |     return ();
   |     --------- not a function or closure

error[E0788]: attribute should be applied to a function definition or closure
  --> $DIR/no-coverage.rs:11:5
   |
LL |     #[coverage(off)]
   |     ^^^^^^^^^^^^^^^^
LL |     const X: u32;
   |     ------------- not a function or closure

error[E0788]: attribute should be applied to a function definition or closure
  --> $DIR/no-coverage.rs:14:5
   |
LL |     #[coverage(off)]
   |     ^^^^^^^^^^^^^^^^
LL |     type T;
   |     ------- not a function or closure

error[E0788]: attribute should be applied to a function definition or closure
  --> $DIR/no-coverage.rs:19:5
   |
LL |     #[coverage(off)]
   |     ^^^^^^^^^^^^^^^^
LL |     fn f(&self);
   |     ------------ not a function or closure

error[E0788]: attribute should be applied to a function definition or closure
  --> $DIR/no-coverage.rs:22:5
   |
LL |     #[coverage(off)]
   |     ^^^^^^^^^^^^^^^^
LL |     fn g();
   |     ------- not a function or closure

error[E0788]: attribute should be applied to a function definition or closure
  --> $DIR/no-coverage.rs:30:5
   |
LL |     #[coverage(off)]
   |     ^^^^^^^^^^^^^^^^
LL |     type T = Self;
   |     -------------- not a function or closure

error[E0788]: attribute should be applied to a function definition or closure
  --> $DIR/no-coverage.rs:33:5
   |
LL |     #[coverage(off)]
   |     ^^^^^^^^^^^^^^^^
LL |     type U = impl Trait;
   |     -------------------- not a function or closure

error[E0788]: attribute should be applied to a function definition or closure
  --> $DIR/no-coverage.rs:41:5
   |
LL |     #[coverage(off)]
   |     ^^^^^^^^^^^^^^^^
LL |     static X: u32;
   |     -------------- not a function or closure

error[E0788]: attribute should be applied to a function definition or closure
  --> $DIR/no-coverage.rs:44:5
   |
LL |     #[coverage(off)]
   |     ^^^^^^^^^^^^^^^^
LL |     type T;
   |     ------- not a function or closure

error: unconstrained opaque type
  --> $DIR/no-coverage.rs:34:14
   |
LL |     type U = impl Trait;
   |              ^^^^^^^^^^
   |
   = note: `U` must be used in combination with a concrete type within the same impl

error: aborting due to 13 previous errors

For more information about this error, try `rustc --explain E0788`.
